client/CServerHandler.cpp

```diff
@@ -551,7 +551,7 @@ void CServerHandler::sendStartGame(bool allowOnlyAI) const
 	catch (const std::exception & e)
 	{
 		showServerError( std::string("Unable to start map! Reason: ") + e.what());
-		return;
+		throw e;
```

Comment:
I would prefer to remove try/catch because of this https://github.com/vcmi/vcmi/issues/2633 rather than fixing unknown scenarios, but not in this PR. Will remove change here